wire-library/wire-runtime/src/jvmMain/kotlin/com/squareup/wire/ProtoAdapter.kt

```diff
@@ -215,5 +216,6 @@ actual abstract class ProtoAdapter<E> actual constructor(
     @JvmField actual val DOUBLE: ProtoAdapter<Double> = commonDouble()
     @JvmField actual val BYTES: ProtoAdapter<ByteString> = commonBytes()
     @JvmField actual val STRING: ProtoAdapter<String> = commonString()
+    @JvmField val DURATION: ProtoAdapter<Duration> = DurationProtoAdapter
```

Comment:
This again will require AGP 4.0 and core library desugaring enabled because it's initialized on class load and `DurationProtoAdapter` eager-initializes as a singleton whose constructor then makes reference to `Duration::class` to invoke the superclass constructor.

I don't think you even need this field. Kotlin users can refer to the object and Java users can call `DurationProtoAdapter.INSTANCE`.